frag-nav/src/main/java/com/ncapdevi/fragnav/FragNavTransactionOptions.java

```diff
@@ -0,0 +1,78 @@
+package com.ncapdevi.fragnav;
+
+import android.support.annotation.AnimRes;
+import android.support.annotation.StyleRes;
+import android.support.v4.util.Pair;
+import android.view.View;
+
+import java.util.List;
+
+/**
+ *
+ */
+
+
+public class FragNavTransactionOptions {
+    List<Pair<View, String>> sharedElements;
+    @FragNavController.Transit
+    Integer transition;
+    @AnimRes
+    Integer enterAnimationId;
+    @AnimRes
+    Integer exitAnimationId;
+    @StyleRes
+    Integer transitionStyle;
+
+    private FragNavTransactionOptions(Builder builder) {
+        sharedElements = builder.sharedElements;
+        transition = builder.transition;
+        enterAnimationId = builder.enterAnimationId;
+        exitAnimationId = builder.exitAnimationId;
+        transitionStyle = builder.transitionStyle;
+    }
+
+    public static Builder newBuilder() {
+        return new Builder();
+    }
+
+    public static final class Builder {
```

Comment:
`addToBackStack` I'm going to leave off for now, since there's no possible way that it could be used (don't have access to the fragment transactions and we're handling all of the backstacks ourselves). Maybe there's a case for adding it later, but right now, it would only add confusing I think. Totally agree on adding all the others though